kratos/geometries/geometry.h

```diff
@@ -836,6 +860,54 @@ class Geometry : public PointerVector<TPointType>
         return normal;
     }
 
+    /**
+    * @brief TO BE DONE
+    */
+    virtual array_1d<double, 3> Normal(array_1d<double, 3>& rResult, IndexType IntegrationPointIndex) const
+    {
+        Normal(rResult, IntegrationPointIndex, mpGeometryData->DefaultIntegrationMethod());
+        return rResult;
+    }
+
+    /**
+    * @brief TO BE DONE
+    */
+    virtual array_1d<double, 3> Normal(array_1d<double, 3>& rResult, IndexType IntegrationPointIndex, IntegrationMethod ThisMethod) const
```

Comment:
```suggestion
    array_1d<double, 3> Normal(const Geometry& rGeom, IndexType IntegrationPointIndex, IntegrationMethod ThisMethod); // free function
```
This function can go outside of the class since it doesn't need to access private data.

Reply:
I disagree.... It is better to be were data is and all data are stored in the geometry.

Reply:
I just followed the structure of the functions for shape functions, gradients, ... . It is only for completeness. The other Normal function is right ahead.

Reply:
I just followed the same structure as for the other functions. Shape functions, Gradients... This was just for completeness.

Reply:
@pooyan-dadvand that's fine and I know this isn't the convention in Kratos. But it is considered good practice. For reasons see http://isocpp.github.io/CppCoreGuidelines/CppCoreGuidelines#Rc-member and https://isocpp.org/blog/2018/02/free-your-functions-klaus-iglberger

Reply:
The rule applies if `normal` function doesn't depends on type of the geometry regardless to if data accessed or not

Reply:
I agree with that reasoning 👍 But we should be more careful with the other since it can really affect code quality.